nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_width.py

```diff
@@ -0,0 +1,684 @@
+from nncf.experimental.torch.nas.bootstrapNAS.elasticity.base_handler import ELASTICITY_HANDLERS_MAP
+from nncf.experimental.torch.nas.bootstrapNAS.elasticity.base_handler import SingleElasticityBuilder
+from nncf.experimental.torch.nas.bootstrapNAS.elasticity.base_handler import SingleElasticityHandler
+from nncf.experimental.torch.nas.bootstrapNAS.elasticity.elasticity_dim import ElasticityDim
+from nncf.experimental.torch.nas.bootstrapNAS.elasticity.filter_reorder import FilterReorderingAlgorithm
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.pruning.filter_pruning.functions import FILTER_IMPORTANCE_FUNCTIONS
+from nncf.torch.pruning.operations import PTElementwisePruningOp
+from nncf.torch.pruning.operations import PT_PRUNING_OPERATOR_METATYPES
+from nncf.torch.pruning.tensor_processor import PTNNCFPruningTensorProcessor
+from nncf.torch.pruning.utils import collect_input_shapes
+from nncf.torch.pruning.utils import collect_output_shapes
+from nncf.torch.tensor import PTNNCFTensor
+from nncf.torch.utils import get_filters_num
+
+pruning_group_id = int
+width_type = int
+ElasticWidthConfig = Dict[pruning_group_id, width_type]
+WidthList = List[width_type]
+ElasticWidthSearchSpace = Dict[pruning_group_id, WidthList]
+
+
+class DynamicWidthOp:
+    """
+    Base class for operations that dynamically adapt the layer parameters after removing some
+    number of filters of preceding layers (input width) or the current layer (output width).
+    """
+
+    def __init__(self, *args, max_width: int, node_name: str, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._active_width = max_width
+        self._max_width = max_width
+        self._node_name = node_name
+
+    @property
+    def max_width(self) -> width_type:
+        return self._max_width
+
+    def get_active_width(self) -> width_type:
+        return self._active_width
+
+    def set_active_width(self, width: width_type):
+        """
+        Set new number of filters that was set for preceding layers (input width) or for current layer
+        (output width). The operation will use this value to adapt parameters for the layer, to which the operation is
+        applied. The width should be less the original number of filters and more than one. Zero number of filters is
+        supported through Dynamic Depth feature.
+
+        :param width: number of filters that was set for preceding layers.
+        """
+        if width is None or width > self._max_width or width < 1:
+            raise AttributeError('Invalid width={} in scope={}.\nIt should be within the range: [1, {}]'.format(
+                width, self._node_name, self._max_width))
+
+        self._active_width = width
+
+
+class ElasticWidthOp(DynamicWidthOp):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._width_list = []
+
+    @property
+    def width_list(self) -> List[int]:
```

Comment:
Docstring would be helpful

Reply:
thanks for the comment! I've added
please review

Reply:
Now it's clear for me, thanks!